sunpy/instr/fermi.py

```diff
@@ -0,0 +1,373 @@
+        vy_primed = np.cross(scz_vector, vx_primed)
+
+        # do the second part of the rotation around vy
+        vz_primed = rotate_vector(scz_vector, vy_primed, np.deg2rad(theta))
+
+        # now we should be pointing at the new RA/DEC.
+        ra = Longitude(np.degrees(math.atan2(vz_primed[1], vz_primed[0])) * u.deg)
+        dec = Latitude(np.degrees(math.asin(vz_primed[2])) * u.deg)
+
+        # save the RA/DEC in a dictionary
+        detector_radecs[l] = [ra, dec]
+
+    detector_radecs['time'] = time
+    return detector_radecs
+
+
+def rotate_vector(vector, axis, theta):
+    # the Euler-Rodrigues formula for rotating vectors
+    # axis is the vector to rotate around
+    # theta is the angle to rotate
+    # http://en.wikipedia.org/wiki/Euler-Rodrigues_parameters#Rotation_angle_and_rotation_axis
+
+    axis = axis/np.sqrt(np.dot(axis, axis))
+    a = np.cos(theta/2)
+    b, c, d = -axis*np.sin(theta/2)
+
+    rot_matrix = np.array([[a*a+b*b-c*c-d*d, 2*(b*c+a*d),2*(b*d-a*c)],
+                     [2*(b*c-a*d), a*a+c*c-b*b-d*d, 2*(c*d+a*b)],
+                     [2*(b*d+a*c), 2*(c*d-a*b), a*a+d*d-b*b-c*c]])      
+
+    return np.dot(rot_matrix,vector)
+
+
+def get_detector_separation_angles(detector_radecs, sunpos):
+    '''Finds the separation angle between the Sun and each NaI detector, given a dictonary of detector RA/DECs.'''
+    angles = copy.deepcopy(detector_radecs)
+    for l, d in detector_radecs.items():
+        if not l == 'time':
+            angle = separation_angle(d, sunpos)
+            angles[l] = angle
+    return angles
+
+
+def separation_angle(radec1, radec2):
+    '''use the law of spherical cosines to calculate the separation angle between two RA/DEC positions.
+     radec1 and radec2 are quantities'''
+    #cosine_of_angle = ( np.cos(np.deg2rad(90 - radec1[1])) * np.cos(np.deg2rad(90 - radec2[1])) ) + ( np.sin(np.deg2rad(90 - radec1[1])) * np.sin(np.deg2rad(90 - radec2[1])) * np.cos(np.deg2rad(radec1[0] - radec2[0])) )
+    
+    cosine_of_angle = ( np.cos( ((90*u.deg) - radec1[1].to('degree')).to('rad')) * np.cos( (90*u.deg - radec2[1].to('degree')).to('rad')) ) + ( np.sin( ((90*u.deg) - radec1[1].to('degree')).to('rad') ) * np.sin( ((90*u.deg) - radec2[1].to('degree')).to('rad') ) * np.cos( (radec1[0].to('degree') - radec2[0].to('degree')).to('rad')   ) )
+
+
+    angle=(np.arccos(cosine_of_angle)).to('degree')
+
+    return angle
+
+
+def met_to_utc(timeinsec):
+    '''Converts Fermi Mission Elapsed Time (MET) in seconds to a datetime object.'''
+    # times for GBM are in Mission Elapsed Time (MET). The reference time for this is 2001-Jan-01 00:00.
+    met_ref_time = parse_time('2001-01-01 00:00')
+    offset_from_utc = (met_ref_time - parse_time('1979-01-01 00:00')) #.total_seconds() #have to do this the long way for python 2.6 compatibility
+    total_offset_secs = (offset_from_utc.microseconds + (offset_from_utc.seconds + offset_from_utc.days * 24 * 3600) * 10**6) / 10**6     
```

Comment:
you can safely use `datetime.total_seconds()` now because we have dropped Python 2.6 support.